### PR TITLE
Add humor appendix

### LINKED ISSUE #163: Rust Humor index
Rust is ... funny :rofl: 

This stuff is perhaps worth an appendix?

- Weird exprs test: https://github.com/rust-lang/rust/blob/master/src/test/run-pass/weird-exprs.rs
- Ferris Rap: http://fitzgeraldnick.com/2018/12/13/rust-raps.html
- The Genesis of Generic Germination: https://github.com/rust-lang/rust/pull/53645#issue-210543221
- The Bastion of the Turbofish test: https://github.com/rust-lang/rust/blob/79d8a0fcefa5134db2a94739b1d18daa01fc6e9f/src/test/ui/bastion-of-the-turbofish.rs

### PR DESCRIPTION
**Pull request by mark-i-m**
Closes #163

r? @spastorino

### CODE REVIEW

**Comment by spastorino** [comment]
I think the idea may be nice to be honest :). On the other hand there are some included things that I think doesn't belong to the compiler explicitly. I'm probably not the best person to check this kind of stuff though :smile:

**Comment by mark-i-m** [comment]
Oops sorry, I missed this. I do see your point... but the forge doesn't seem any better suited, and I can't think of anywhere else to host this.

**Comment by spastorino** [comment]
@mark-i-m yeah, consider my position to this neutral :smile:. If you want, go ahead and merge it, otherwise let's see what others say?.

**Comment by mark-i-m** [comment]
Nobody else seems to have commented here after 2 weeks, so I will go ahead and merge it.

If a future traveler has opinions, feel free to leave them.

**Comment by RalfJung** [comment]
Shouldn't that also mention `break rust;`? ;)

**Comment by mark-i-m** [comment]
@RalfJung What is `break rust;`? I will gladly add it if you provide a link :)

**Comment by RalfJung** [comment]
I am trying to find it again...

**Comment by RalfJung** [comment]
[Found it](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0ab2bd6a9d722e0f05a95e2a5dcf89cc).

File: src/SUMMARY.md [hunk 1]
```diff
@@ -93,4 +93,9 @@
 [Appendix C: Glossary](./appendix/glossary.md)
 [Appendix D: Code Index](./appendix/code-index.md)
 [Appendix E: Bibliography](./appendix/bibliography.md)
+
+[Appendix Z HumorRust](./appendix/humorust.md)
```
Comment (Phrohdoh) [inline src/SUMMARY.md]: Looks like there should have been a `:` after the `Z`.

Comment (mark-i-m) [reply src/SUMMARY.md]: @Phrohdoh Yep, this was fixed recently.

**Pull request merged**